---
title: "**Data Files Editor**"
output: html_document
---

### **Overview**

This file is used to make edits to the data files. 

```{r,echo=FALSE,results='hide',message=FALSE, warning=FALSE}
#Load DT, dplyr
library(DT)
#library(dplyr) 
library(here)
library(tidyverse)
library(magrittr)
#library(RCurl)
```

##  Resource table editor
```{r,echo=FALSE,results='hide',message=FALSE, warning=FALSE}
# Load file for Resource Table
resource <- readr::read_csv(file = here::here("data/itcr_table_resources.csv"))

tibbleTOPAS <- tibble(
  Name = "TOPAS",
  Price = '<a href="http://www.topasmc.org/licensing" style="color: #be3b2a" target="_blank"<div title="Pricing Link"> </div>Paid/Free</a>'
)

resource <- dplyr::rows_update(resource, tibbleTOPAS, by="Name") #this will update the existing TOPAS entry

## Add ISIC Archive: Submitted 06/01/2023, 13:31 (vrotemberg@gmail.com)
new_ISIC <- tibble(
  Name = "ISIC Archive",
  Funding = "Yes",
  Subcategories = "Cancer Imaging", 
  `Data Type` = "Clinical imaging",
  `Unique Factors` = "The largest public archive of dermatology photographs. It has around 20 million individual image downloads per month and is widely used for education and AI research.",
  Price = "Free",
  Link = "https://www.isic-archive.com/",
  `Summary` = "The ISIC Archive is an academic-industry collaboration that is devoted to improving dermatology artificial intelligence. It is the largest public archive of images of skin conditions, with special attention to skin cancer detection. We also host grand challenges (and live challenges) for AI models as well as conference workshops.")

resource <- dplyr::rows_insert(resource, new_ISIC, conflict = "ignore") #include conflict = ignore argument because it won't re-add the resource the next time this code is run.

## Goal: Create the html version of links in resources
## Edit 1: Add a column "identifier" to the resource table to use it as a merge key variable with the Data provided column entries in other tables
## Edit 2: Make sure add new identifiers for new resources in the resource data set.
resource <- resource %>% 
  mutate(identifier = c("PDX", "HemOnc", "ngchm", "OncoMX", "MethCon5", "CIViC", "BioMuta", "TCIA", "SCGV", "ivyGlimpse", "TCPA", "dcmqi", "Region Templates", "Cistrome", "TOPAS", "TCGA", "1000 Genomes", "TARGET", "CCLE", "dbGAP", "CPTAC", "PheKB", "NSCLC Radiomics", "ICGC", "ISIC"))  %>%  relocate(identifier)

## From this point forward, any additions to the resource table need to include an "identifier" in the tibble/row that will be added

## Add The Cancer Epitope Database and Analysis Resource (CEDAR): Submitted 07/26/2023, 10:48 (nblazeska@lji.org)
new_CEDAR <- tibble(
  Name = "Cancer Epitope Database and Analysis Resource (CEDAR)",
  Funding = "Yes", #Not certain about this
  Subcategories = "Proteomics" , #this is my assumption based on the use of epitopes; it was not submitted in the form or explicitly said on the website
  `Data Type` = "Curated literature data: experimentally-tested epitopes and their immunological, biological, and clinical contexts" ,
  `Unique Factors` = "It is a free resoure that provides the cancer community with a central repository of experimentally-tested cancer epitopes curated from peer-reviewed literature.",
  Price = "Free",
  Link = "https://cedar.iedb.org/",
  Summary = "The Cancer Epitope Database and Analysis Resource (CEDAR) is a freely available resource funded by NCI. It catalogs experimental data on antibody and T cell epitopes studied in humans, non-human primates, and other animal species in the context of cancer disease. CEDAR also hosts tools to assist in the prediction and analysis of cancer epitopes." ,
  identifier = "CEDAR"
)

## Add ARCHS4: Submitted 09/13/2023, 12:36 (avi.maayan@mssm.edu)
new_ARCHS4 <- tibble(
  Name = "ARCHS4",
  Funding = "Yes", #Fairly certain about this
  Subcategories = "Transcriptomics",
  `Data Type` = "Transcriptomics, GEO",
  `Unique Factors` = "This resource is updated frequently. It contains all GEO samples and includes metadata and data search.",
  Price = "Free for academic institutions",
  Link = "https://maayanlab.cloud/archs4/" ,
  Summary = "ARCHS4 provides gene and trasncript counts uniformly processed from all the human and mouse RNA-seq samples from GEO" ,
  identifier = "ARCHS4"
)


resource <- dplyr::rows_insert(resource, dplyr::bind_rows(new_CEDAR, new_ARCHS4), conflict = "ignore")


## Save the edited data file
write.csv(resource_identifier, file = "data/itcr_table_resources_identifier.csv") #Do I want to just rewrite the original file?
```

## Adding new platforms to the Imaging Table

```{r,echo=FALSE,results='hide',message=FALSE, warning=FALSE}
imaging <- read_csv(file = here::here("data/ictr_table_imaging.csv"))
# nameOfCols <- imaging %>% colnames()

## CaPTk: Submitted 06/01/2023, 11:12 (sbakas@upenn.edu)
new_imaging_CaPTk <- tibble(
  Name = "The Cancer Imaging Phenomics Toolkit (CaPTk)",
  `Link (Hyperlinked Over Name)` = "https://www.med.upenn.edu/cbica/captk/",
  Subcategories = "Radiomics", 
  `Unique Factor` = "CaPTk is a stand-alone software client that allows for free image analysis through the Image Processing Portal (IPP) that offers compute power through CaPTk's supporting high performance computing cluster.",
  `Data Types (Clean)` = "Radiographic Imaging",
  Price = "Free",
  Documentation = "https://cbica.github.io/CaPTk/",
  `Data Provided` = "Sample Data available within CaPTk",
  Publications = "https://doi.org/10.1117/1.jmi.5.1.011018",
  `Summary` = "CaPTk is a software platform for analysis of radiographic cancer images, currently focusing on brain, breast, and lung cancer that integrates advanced, validated tools performing various aspects of medical image analysis, that have been developed in the context of active clinical research studies and collaborations toward addressing real clinical needs. CaPTk aims to facilitate the swift translation of advanced computational algorithms into routine clinical quantification, analysis, decision making, and reporting workflow.",
  Funding = "Yes",
  PLink = "https://github.com/CBICA/CaPTk/#downloads")

## Add new rows to the existing dataset
new_imaging <- dplyr::rows_insert(imaging, new_imaging_CaPTk, conflict = "ignore")

## Create new data file 
write.csv(new_imaging, file = "data/ictr_table_imaging_new.csv", row.names = FALSE) #Do I want to just rewrite the original file?
```


